### ISSUE TITLE
Negative TimeSpan result when hours are 0 is converted to positive

### ISSUE THREAD

**Issue by euantorano** [mysql-net/MySqlConnector#518]
When running a query that returns a `TimeSpan` value, if the value is a negative value less than 1 hour, the timespan is converted into a positive.

For example, take the following query:

```sql
SELECT TIMEDIFF(DATE_SUB(NOW(), INTERVAL 5 MINUTE), NOW()) AS span;
```

The expected result would be: `-00:05:00`

Now, when running that query with the following code, the value is actually `00:05:00`:

```c#
using (MySqlConnection connection = new MySqlConnection(connectionString))
{
    connection.Open();

    const string query = @"SELECT TIMEDIFF(DATE_SUB(NOW(), INTERVAL 5 MINUTE), NOW()) AS span;";

    using (MySqlCommand cmd = new MySqlCommand(query, connection))
    {
        object result = cmd.ExecuteScalar();

        if (result is TimeSpan ts)
        {
            Console.WriteLine($"Got timespan result: {ts}");
        }
        else
        {
            Console.WriteLine($"Got result of invalid type '{result.GetType().Name}': {result}");
        }
    }
}
```

I've tracked this issue down to the following line of code: https://github.com/mysql-net/MySqlConnector/blob/d6288a24e625a3c911eba36a17e295903398506a/src/MySqlConnector/Core/Row.cs#L534

Looking at the source for `Utf8Parser.TryParse`, it looks like if the value is 0, the sign in front will be lost (eg: `-0` becomes `0`): https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Number.cs#L17

Looking at `Utf8Parser.TryParse(ReadOnlySpan<byte> source, out TimeSpan value, out int bytesConsumed, char standardFormat = default))`, they handle any sign at the start before parsing any further: https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.TimeSpan.BigG.cs#L28

I wonder if it's possible for `MySqlConnector` to simple use `Utf8Parser.TryParse(ReadOnlySpan<byte> source, out TimeSpan value, out int bytesConsumed, char standardFormat = default))`?

**Comment by bgrainger** [comment]
> I wonder if it's possible for MySqlConnector to simple use `Utf8Parser.TryParse(ReadOnlySpan<byte> source, out TimeSpan value, out int bytesConsumed, char standardFormat = default))`?

Yes, it looks like this is possible, and would fix the bug.

**Comment by bgrainger** [comment]
Fixed in [0.42.1](https://github.com/mysql-net/MySqlConnector/releases/tag/0.42.1).
